sentry-android-core/src/main/java/io/sentry/android/core/gestures/ViewUtils.java

```diff
@@ -0,0 +1,101 @@
+package io.sentry.android.core.gestures;
+
+import android.content.res.Resources;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AbsListView;
+import android.widget.ScrollView;
+import androidx.core.view.ScrollingView;
+import io.sentry.util.Objects;
+import java.util.ArrayDeque;
+import java.util.Queue;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class ViewUtils {
+  static @Nullable View findTarget(
+    final @NotNull View decorView,
+    final float x,
+    final float y,
+    final @NotNull ViewTargetSelector viewTargetSelector
+  ) {
+    Queue<View> queue = new ArrayDeque<>();
+    queue.add(decorView);
+
+    @Nullable View target = null;
+    // the coordinates variable can be method-local, but we allocate it here, to avoid allocation
+    // in the while- and for-loops
+    int[] coordinates = new int[2];
+
+    while (queue.size() > 0) {
+      final View view = Objects.requireNonNull(queue.poll(), "view is required");
+
+      if (viewTargetSelector.select(view)) {
+        target = view;
+      }
+
+      if (view instanceof ViewGroup) {
+        final ViewGroup viewGroup = (ViewGroup) view;
+        for (int i = 0; i < viewGroup.getChildCount(); i++) {
+          final View child = viewGroup.getChildAt(i);
+          if (touchWithinBounds(child, x, y, coordinates)) {
+            queue.add(child);
+          }
+        }
+      }
+    }
+
+    return target;
+  }
+
+  private static boolean touchWithinBounds(
+    final @NotNull View view,
+    final float x,
+    final float y,
+    final int[] coords
+  ) {
+    view.getLocationOnScreen(coords);
+    int vx = coords[0];
+    int vy = coords[1];
+
+    int w = view.getWidth();
+    int h = view.getHeight();
+
+    return !(x < vx || x > vx + w || y < vy || y > vy + h);
+  }
+
+  static boolean isViewTappable(final @NotNull View view) {
+    return view.isClickable() && view.getVisibility() == View.VISIBLE;
+  }
+
+  static boolean isViewScrollable(final @NotNull View view) {
```

Comment:
in theory we could remove this, and then we don't need to add support for anything that I mentioned in the PR description, but then if you move a finger over a static view (say, textview/button/etc.), then the breadcrumb will be also captured as `swipe`. Open for ideas, but for now don't see any other way.

Reply:
I'm afraid of doing reflection on every scroll call, that might cause junk in the UI

Reply:
We're only doing this on the first onScroll occurrence, so should be alright?

```Java
if (scrollState.type == null) {
  @SuppressWarnings("Convert2MethodRef")
  final @Nullable View target =
    ViewUtils.findTarget(decorView, firstEvent.getX(), firstEvent.getY(),
      view -> ViewUtils.isViewScrollable(view));

  scrollState.setTarget(target);
  scrollState.type = "scroll";
}
```

Reply:
So is this being called only once? during the whole App's lifecycle? by reading the code it looks like its being called quite often, but I didn't execute the code.

Reply:
This is called when a user starts scrolling/swiping everytime. But only once, in the beginning, we identify the targetView and store it until the ACTION_UP is performed (that's when we capture a breadcrumb and nullify the target view ref). Next time, the user starts scrolling again, then we again call this method to identify the view target.

Reply:
Is there any other way? Did you check how others do it? If not, can we benchmark with and w/o it? I'm afraid that this would be a problem for low end devices.

Reply:
Yeah others do it similarly. I don't think there's any other way to find a target view, you essentially need to traverse the tree to find out.  

Actually, while writing this, I realized we can improve this for the scrollable case, we can return early if the `viewTargetSelector.select` returned true (so we don't need to traverse children of the ScrollView/RecyclerView, if we just hit the ScrollView). This way I think there won't be many iterations for the while-loop, because the scrollviews are usually somewhere on the top of the view hierarchy.

Another improvement - we can pass the androidXAvailable boolean as a parameter, so we don't check `Class.forName` every time, but somewhere once, when the integration gets installed for instance.

As for benchmarking, we could use [macrobencmark lib](https://github.com/android/performance-samples/blob/main/MacrobenchmarkSample/macrobenchmark/src/main/java/com/example/macrobenchmark/FrameTimingBenchmark.kt), but it supports only API 23 and above, and needs to run on a real device (the emulator won't give us much confidence probably)

Reply:
I believe that the first two optimizations would do for now.